
Added a check for gcc supporting -fno-unit-at-a-time and tidied up some
whitespace in arch/um/Makefile.
Added -DUM_FASTCALL to Makefile-i386.
make clean descends into util in order to get rid of the binaries there.

Signed-off-by: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
---

 arch/um/util/Makefile                             |    0 
 linux-2.6.9-current-paolo/arch/um/Makefile        |    2 +-
 linux-2.6.9-current-paolo/arch/um/Makefile-i386   |    5 +++++
 linux-2.6.9-current-paolo/arch/um/kernel/Makefile |    5 +++++
 4 files changed, 11 insertions(+), 1 deletion(-)

diff -puN arch/um/kernel/Makefile~uml-makefile-fixes arch/um/kernel/Makefile
--- linux-2.6.9-current/arch/um/kernel/Makefile~uml-makefile-fixes	2004-09-18 16:17:28.416972256 +0200
+++ linux-2.6.9-current-paolo/arch/um/kernel/Makefile	2004-09-18 16:17:28.421971496 +0200
@@ -5,6 +5,11 @@
 
 extra-y := vmlinux.lds uml.lds dyn.lds
 
+# Descend into ../util for make clean.  This is here because it doesn't work
+# in arch/um/Makefile.
+
+subdir- = ../util
+
 obj-y = checksum.o config.o exec_kern.o exitcode.o frame_kern.o frame.o \
 	helper.o init_task.o irq.o irq_user.o ksyms.o main.o mem.o mem_user.o \
 	physmem.o process.o process_kern.o ptrace.o reboot.o resource.o \
diff -puN arch/um/Makefile~uml-makefile-fixes arch/um/Makefile
--- linux-2.6.9-current/arch/um/Makefile~uml-makefile-fixes	2004-09-18 16:17:28.417972104 +0200
+++ linux-2.6.9-current-paolo/arch/um/Makefile	2004-09-18 16:19:10.798407912 +0200
@@ -69,7 +69,7 @@ LINK_WRAPS = -Wl,--wrap,malloc -Wl,--wra
 # included; the values here are meaningless
 
 CONFIG_NEST_LEVEL ?= 0
-CONFIG_KERNEL_HALF_GIGS ?=  0
+CONFIG_KERNEL_HALF_GIGS ?= 0
 
 SIZE = (($(CONFIG_NEST_LEVEL) + $(CONFIG_KERNEL_HALF_GIGS)) * 0x20000000)
 
diff -puN arch/um/Makefile-i386~uml-makefile-fixes arch/um/Makefile-i386
--- linux-2.6.9-current/arch/um/Makefile-i386~uml-makefile-fixes	2004-09-18 16:17:28.418971952 +0200
+++ linux-2.6.9-current-paolo/arch/um/Makefile-i386	2004-09-18 16:17:28.422971344 +0200
@@ -5,6 +5,11 @@ TOP_ADDR = 0xc0000000
 endif
 
 CFLAGS += -U__$(SUBARCH)__ -U$(SUBARCH)
+
+ifneq ($(CONFIG_GPROF),y)
+ARCH_CFLAGS += -DUM_FASTCALL
+endif
+
 ELF_ARCH = $(SUBARCH)
 ELF_FORMAT = elf32-$(SUBARCH)
 
diff -puN arch/um/util/Makefile~uml-makefile-fixes arch/um/util/Makefile
_
